advanced_alchemy/operations.py

```diff
@@ -1,10 +1,15 @@
 from __future__ import annotations
 
+import sys
 from typing import TYPE_CHECKING, Any
 
 from sqlalchemy import ClauseElement, ColumnElement, UpdateBase
 from sqlalchemy.ext.compiler import compiles
 
+if sys.version_info < (3, 11):
+    from typing_extensions import Self
+else:
+    from typing import Self
```

Comment:
And so given that `typing_extensions` is already in your environment, you could import this directly from there in all cases and let them handle this conditional. E.g, their source for `Self` is [this](https://github.com/python/typing_extensions/blob/94bec447d6f7b9d3625ef0e688a0b0f9e487e951/src/typing_extensions.py#L1976-L1994):

```py


if hasattr(typing, "Self"):  # 3.11+
    Self = typing.Self
else:
    @_SpecialForm
    def Self(self, params):
        """Used to spell the type of "self" in classes.

        Example::

          from typing import Self

          class ReturnsSelf:
              def parse(self, data: bytes) -> Self:
                  ...
                  return self

        """

        raise TypeError(f"{self} is not subscriptable")
```